packages/api-request-builder/test/create-service.spec.js

```diff
@@ -88,61 +89,283 @@ describe('createService', () => {
     )
   })
 
+  describe('parse', () => {
+    let service
+    beforeEach(() => {
+      const options = {
+        type: 'foo',
+        endpoint: '/foo',
+        features: [
+          defaultFeatures.query,
+          defaultFeatures.queryOne,
+          defaultFeatures.queryExpand,
+          defaultFeatures.projection,
+        ],
+      }
+      service = createService(options, projectKey)
+    })
+
+    describe('features of parsed object', () => {
+      // query-expand
+      it('should support single `expand`', () => {
+        expect(service.parse({ expand: ['bar'] }).build())
+          .toBe('/my-project1/foo?expand=bar')
+      })
+      it('should support mutliple `expand`s', () => {
+        expect(service.parse({ expand: ['bar', 'baz'] }).build())
+          .toBe('/my-project1/foo?expand=bar&expand=baz')
```

Comment:
The tests I added were written against the public API exclusively.

I got this from Dan

> **For libraries, tests should ideally be written against the public APIs.** This goes against the common mantra of unit testing, but in our experience this helps both ensure that the right thing is being tested, and that it is easy to replace the underlying implementation. It does, however, make debugging test failures harder. - [source](https://hashnode.com/post/what-have-you-learned-after-working-at-facebook-for-almost-two-years-have-you-grown-as-a-developer-and-what-are-some-of-the-key-takeaways-cj7q3gkjx019xkhwujchsrtho)